ch02.adoc

```diff
@@ -163,5 +163,76 @@ The global **`external_variables`** attribute is a blank-separated list of the n
 These variables are to be found in other files (called "external files") but CF does not provide conventions for identifying the files concerned.
 The only attribute for which CF standardises the use of external variables is **`cell_measures`**.
 
+[[groups, Section 2.7, "Groups"]]
+=== Groups
 
+Groups provide a powerful mechanism to structure data hierarchically.
+This convention does not standardize group names.
+It may be of benefit to name groups in such a way that human readers can interpret them.
+However, files that conform to this standard shall not require software to interpret or decode information from group names.
+References to out-of-group variable and dimensions shall be found by applying the scoping rules outlined below.
 
+==== Scope
+
+Any variable or dimension can be referred to, as long as it can be found with one of the following search strategies:
+
+* Search by absolute path
+* Search by relative path
+* Search by proximity
+
+These strategies are explained in detail in the following sections.
+
+If any dimension of an out-of-group variable has the same name as a dimension of the referring variable, the two must be the same dimension (i.e. they must have the same netCDF dimension ID).
+
+===== Search by absolute path
+
+A variable or dimension specified with an absolute path (i.e., with a leading slash "/") is at the indicated location relative to the root group, as in a UNIX-style file convention.
+For example, a `coordinates` attribute of `/g1/lat` refers to the `lat` variable in group `/g1`.
+
+===== Search by relative path
+
+As in a UNIX-style file convention, a variable or dimension specified with a relative path (i.e., containing a slash but not with a leading slash, e.g. `child/lat`) is at the location obtained by affixing the relative path to the absolute path of the referring attribute.
+For example, a `coordinates` attribute of `g1/lat` refers to the `lat` variable in subgroup `g1` of the current (referring) group.
+Upward path traversals from the current group are indicated with the UNIX convention.
+For example, `../g1/lat` refers to the `lat` variable in the sibling group `g1` of the current (referring) group.
+
+===== Search by proximity
+
+A variable or dimension specified with no path (for example, `lat`) refers to the variable or dimension of that name, if there is one, in the referring group.
+If not, the ancestors of the referring group are searched for it, starting from the direct ancestor and proceeding toward the root group, until it is found.
+
```

Comment:
Please give an example so we can be sure to address your concern, which is unclear to me.

Reply:
Pinging @marqh to please give an example, preferably in CDL, of the situation about which you think a note should be added.

Reply:
Hello @czender 

I'd like to reuse the examples that I think should be included in this PR, but are currently missing

do you know if thiis interpretation is correct, is there a file missing from this PR, which includes example CDL with groups?

thank you
mark

Reply:
I am unsure whether Appendix I will be in the initial PR. Please read the following exchange between @JonathanGregory and @erget (from #144):

>> Could we please omit Appendix I for the moment from the pull request? By the way, it is entitled Appendix A.

> I don't really understand this request, to be honest, because I think that it would make it easier to interpret the additions, but I've done it for the moment ([fcad2b7](https://github.com/cf-convention/cf-conventions/commit/fcad2b7cddc264b2ccb7bf38876d0a3b4f5a5257)). Also, the reason it's called "Appendix A" in the rendering is because the appendix numbering actually takes place during the compilation of the Conventions as a whole, so when you view the individual file, the `[appendix]` tag is converted into "Appendix A" due to the lack of context. If you compile the complete Conventions it's titled appropriately.
> 
> I don't think that the appendix is really a break from convention, especially in light of Appendix H, which is large and comprehensive, as well as extremely useful, and would prefer to keep it, but I'll defer to your judgement.

Reply:
I also think that the appendix adds significant informational value and should be included.

Not withstanding that opinion, I think that this PR should be self consistent. That suggests to me either:
* re-include the appendix
* remove references to the appendix from this PR

My preference is to re-include the appendix.

If @JonathanGregory has particular reasons why not including the appendix is preferred, then I'd be interested to hear them. @JonathanGregory please may you share your view with us here?  thank you.

If non-inclusion is the prevailing view, then I think it preferable that the references are removed so that a merged PR is self-consistent.

Reply:
@JonathanGregory would you be willing to weigh in on this? As stated in #144 I do agree with @marq here that it would be better to include the appendix. My opinion is also that any issues that we have with an appendix can be addressed more easily should we find them, as the appendix is consistent with the proposed changes to chapter 2, so the only issues should be editorial.

@marqh in the meantime I'm removing the references in order to keep with the discussion in #144, but this is isolated to 664e10f so we can revert the commit later if needed. As stated in previous discussions, I am *in favour* of keeping Appendix I and agree with your assertion that it adds informational value.

Reply:
thank you for following this through, 

the change, for consistency, is sensible, but the preference from me would be to revert this change and reintroduce the appendix, as I have previously stated.

however, this change is sufficient to remove my objection

Reply:
Hello, I'm sorry if this has been dealt with elsewhere, but it looks to me as though the [NASA EOS file format convention](https://earthdata.nasa.gov/user-resources/standards-and-references/hdf-eos5) could break the proposed lateral search algorithm. As such EO datasets have been advanced as a reason for including support for implicit references outside the direct parental line of a group, this would be a problem. 
It may be that I have misunderstood the term "local apex group": is this defined anywhere?

Reply:
@martinjuckes The proposed CF lateral search algorithm has evolved over time from a stronger (and more complex and fragile) form that did support most NASA EOS datasets to a weaker (and less complex and fragile) form that will not support many NASA EOS datasets. Perhaps the most relevant change is that the lateral search algorithm now proposed is restricted to single-dimension NUG-style coordinates, rather than the multidimensional auxiliary coordinates originally proposed. Please see #144 for more discussion. I would characterize this as the proposed CF convention not supporting NASA EOS rather than NASA EOS breaking the proposed convention. The NASA EOS convention predates the proposed CF Groups convention by many years. 
This PR defines the term "local apex group" is defined in the ch01.adoc glossary as

>  "local apex group:: The nearest (to a referring group) ancestor group in which a dimension underlying an out-of-group coordinate is defined. The word "apex" refers to position of this group at the vertex of the tree of groups formed by it, the referring group, and the group where a coordinate is located."